### DISCUSSION TITLE
Leave method is not executed on useTransition

### DISCUSSION THREAD

**Discussion by prrrcl** [pmndrs/react-spring#1629] [Support]
hi, i did a replica of my code on codesandbox. 

I added a component with the useTransition hook inside. It depends on location for "animation management", like router.
But i dont get any response about leave method :S 

Why ?

https://codesandbox.io/s/infallible-northcutt-0396k?file=/src/pages/Home.js

**Comment by Sh4reef** [comment]
@prrrcl I've checked your codesandbox, it seems the leave method works fine for me.

**Reply by prrrcl** [reply]
![simpleAnim.js](https://user-images.githubusercontent.com/44324243/125767600-beb152fc-873e-43ef-86d8-2f8b5b2621df.png)
Sorry, the file that I told you is: SimpleAnim.js. There is console.log on Leave method, and it doesn't show anything on console.

**Comment by Sh4reef** [comment]
@prrrcl I tried to comment out the `keys` prop line `17` and it seems working, but only when you click the links quickly.

**Reply by prrrcl** [reply]
How can i fix that? I mean, i hope to do a transition enter and leave every pathname changes, how can i do that?

**Reply by somewonderfulguy** [reply]
It works. But sometimes. Really. Not reliable. In v8 `leave` was pretty much reliable - always worked.
